internal/expr/bindtypes.go

```diff
@@ -49,69 +49,97 @@ func (pe *ParsedExpr) BindTypes(args ...any) (tbe *TypeBoundExpr, err error) {
 	// Bind types to each expression.
 	var typedExprs TypeBoundExpr
 	outputUsed := map[string]bool{}
-	var te any
 	for _, expr := range pe.exprs {
-		switch e := expr.(type) {
-		case *inputExpr:
-			te, err = bindInputTypes(e, argInfo)
-			if err != nil {
-				return nil, err
-			}
-		case *outputExpr:
-			toe, err := bindOutputTypes(e, argInfo)
-			if err != nil {
-				return nil, err
-			}
+		te, err := expr.bindTypes(argInfo)
+		if err != nil {
+			return nil, err
+		}
 
+		if toe, ok := te.(*typedOutputExpr); ok {
```

Comment:
This is OK for now, but we should keep an eye out for a way to poke it into the abstraction you've created.

Reply:
Agreed, this chunk sticks out like a sore thumb.